stl/inc/xstring

```diff
@@ -118,7 +118,7 @@ struct _Char_traits { // properties of a string or stream element
         if (!_STD is_constant_evaluated())
 #endif // __cpp_lib_is_constant_evaluated
         {
-            return reinterpret_cast<_Elem*>(_CSTD memset(reinterpret_cast<wchar_t*>(_First), _Ch, _Count));
+            return reinterpret_cast<_Elem*>(_CSTD memset(_First, _Ch, _Count));

```

Comment:
```suggestion
            return static_cast<_Elem*>(_CSTD memset(_First, _Ch, _Count));

```

Reply:
Pre-existing: How is this not completely broken when `sizeof(_Elem) != 1`?

Reply:
...unless I can convince everyone that we should not ship a primary `char_traits` template?

Reply:
Ist there a specialization for `wchar_t`?

I guess that should be safe then

Reply:
Argh, this was just me bein clueless and assuming that memset would to the right thing.
Turns out it requires jts second argument (`_Ch`) to be of type `int` and then casts it hard to `unsigned char`. 

That means we cannut use it here. 

And I was wondering why it was missing...